src/build.rs

```diff
@@ -310,23 +318,29 @@ impl Build {
 
         if !self.disable_zig_linker {
             // setup zig as linker
-            if let Some(target) = self.target.as_ref() {
-                let rustc_meta = rustc_version::version_meta()?;
-                let host_target = &rustc_meta.host;
+            let rustc_meta = rustc_version::version_meta()?;
+            let host_target = &rustc_meta.host;
+            for (i, full_target) in rust_targets.iter().enumerate() {
                 // we only setup zig as linker when target isn't exactly the same as host target
-                if host_target != target {
```

Comment:
Here before the change this `target` can be `x86_64-unknown-linux-gnu.2.17`, but the changed `full_target` doesn't preserve the GLIBC version part I think?

> // we only setup zig as linker when target isn't exactly the same as host target

When `host_target` is x86_64-unknown-linux-gnu while the `--target` is `x86_64-unknown-linux-gnu.2.17` we need to setup zig as linker even if the only difference is glibc version.

Reply:
>                 if host_target != full_target {

So you want to preserve the version? so `target` will be `x86_64-unknown-linux-gnu.2.17`?

Reply:
Yes, otherwise you can't target lower glibc versions on a high glibc version OS.

Reply:
Added a test to ensure the build artifact can run on CentOS 7: https://github.com/messense/cargo-zigbuild/commit/82ee218ca5194db185ba1f0168d05d17dc09b7fb , please do a rebase to include it.